sqlobject/src/main/java/org/jdbi/v3/sqlobject/DefaultMethodHandler.java

```diff
@@ -18,29 +18,55 @@
 import java.lang.invoke.MethodHandle;
 import java.lang.invoke.MethodHandles;
 import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationTargetException;
 import java.lang.reflect.Method;
 import java.util.Map;
 import java.util.WeakHashMap;
 
 import org.jdbi.v3.core.extension.HandleSupplier;
 
 class DefaultMethodHandler implements Handler {
+    // MethodHandles.privateLookupIn(Class, Lookup) was added in JDK 9.
+    // JDK 9 allows us to unreflectSpecial() on an interface default method, where JDK 8 did not.
+    private static final Method PRIVATE_LOOKUP_IN = privateLookupIn();
+
+    private static Method privateLookupIn() {
+        try {
+            return MethodHandles.class.getMethod("privateLookupIn", Class.class, MethodHandles.Lookup.class);
+        } catch (NoSuchMethodException e) {
+            // Method was added in JDK 9
+            return null;
+        }
+    }
+
     private static final Map<Class<?>, MethodHandles.Lookup> privateLookups = synchronizedMap(new WeakHashMap<>());
 
     private static MethodHandles.Lookup lookupFor(Class<?> clazz) {
-        return privateLookups.computeIfAbsent(clazz, type -> {
+        if (PRIVATE_LOOKUP_IN != null) {
             try {
-                // TERRIBLE, HORRIBLE, NO GOOD, VERY BAD HACK
-                // Courtesy of:
-                // https://rmannibucau.wordpress.com/2014/03/27/java-8-default-interface-methods-and-jdk-dynamic-proxies/
+                return (MethodHandles.Lookup) PRIVATE_LOOKUP_IN.invoke(null, clazz, MethodHandles.lookup());
+            } catch (IllegalAccessException | InvocationTargetException e) {
+                String message = String.format(
+                        "Error invoking MethodHandles.privateLookupIn(%s.class, MethodHandles.lookup()) in JDK 9+ runtime",
+                        clazz);
+                throw new RuntimeException(message, e);
```

Comment:
Should we attempt to fall through to the JDK 8 hack here?

Reply:
I'm not sure it would help - if privateLookupIn exists then the java 8 hack won't likely work - unless Java 10+ is different...

Reply:
This is one of those "this should never happen" scenarios. Maybe we should keep the exception so we hear about it if our assumption turns out to be invalid.

Reply:
the MethodHandler performance is  low

Reply:
You observed a slowdown in the latest change?

Reply:
Also, 'slow' can mean a lot of different things.  Even if the method invocation took 10us (crazy long) it still pales to the cost of a roundtrip and query on most DBs...